connectors/src/connectors/notion/temporal/cast_known_errors.ts

```diff
@@ -0,0 +1,37 @@
+import { APIErrorCode, APIResponseError } from "@notionhq/client";
+import {
+  ActivityExecuteInput,
+  ActivityInboundCallsInterceptor,
+  Next,
+} from "@temporalio/worker";
+
+export class NotionCastKnownErrorsInterceptor
+  implements ActivityInboundCallsInterceptor
+{
+  async execute(
+    input: ActivityExecuteInput,
+    next: Next<ActivityInboundCallsInterceptor, "execute">
+  ): Promise<unknown> {
+    try {
+      return await next(input);
+    } catch (err: unknown) {
+      if (APIResponseError.isAPIResponseError(err)) {
+        if (err.code === APIErrorCode.RateLimited) {
+          throw {
+            __is_dust_error: true,
+            message: err.message,
+            type: "notion_rate_limited",
+          };
+        }
+        if (err.code === APIErrorCode.InternalServerError) {
```

Comment:
What are these errors?

Reply:
The real meta question, is should we page ourselves if they happen?

Reply:
The ones we regularly have are transient internal server errors that on the search endpoint for big workspaces. Technically, they could be anything

Reply:
OK if regular, we should not page ourselves and therefore this is good to go :+1: